app/lib/audit/models.dart

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../account/models.dart';
+import '../shared/datastore.dart' as db;
+import '../shared/utils.dart' show createUuid;
+
+@db.Kind(name: 'AuditLogRecord', idType: db.IdType.String)
+class AuditLogRecord extends db.ExpandoModel<String> {
+  @db.DateTimeProperty(required: true)
+  DateTime created;
+
+  @db.DateTimeProperty()
+  DateTime expires;
+
+  @db.StringProperty(required: true)
+  String kind;
+
+  @db.StringProperty(required: true)
+  String agent;
+
+  @db.StringProperty(required: true)
+  String summary;
+
+  @db.StringListProperty()
+  List<String> users;
```

Comment:
```suggestion
  /// List of [User.id] for the users involved in this record.
  ///
  /// This is the list of users who should have this record show up in their personal audit-log.
  /// This should include [agent], as well as any users who is subject of this action, that's
  /// it should include the user who did this action, and the user who is invited/removed.
  ///
  /// This property **MUST** only be used for querying. The contents of this property is
  /// **NOT** intended for consumption or presentation by other means. It's only here
  /// to make it easy to find [AuditLogRecord]s relevant for a specific user.
  @db.StringListProperty(required: true)
  List<String> users;
```

Any reason not to require this be at-least an empty-list?

TODO: Double check that `[User.id]` is the correct identifier.